query-engine/connectors/sql-query-connector/src/query_builder/write.rs

```diff
@@ -155,10 +156,16 @@ pub fn create_relation_table_records(
     child_ids: &[RecordProjection],
 ) -> Query<'static> {
     let relation = field.relation();
-    let parent_columns: Vec<_> = field.related_field().m2m_columns();
-    let child_columns: Vec<_> = field.m2m_columns();
 
-    let columns: Vec<_> = parent_columns.into_iter().chain(child_columns).collect();
+    let parent_columns = field
+        .related_field()
+        .m2m_columns()
+        .into_iter()
+        .map(|c| c.name.into_owned());
+
+    let child_columns = field.m2m_columns().into_iter().map(|c| c.name.into_owned());
+
+    let columns: Vec<_> = parent_columns.chain(child_columns).collect();
```

Comment:
These changes here above are just different code, no real difference in logic, right?

Reply:
I need to backtrack these. We do it in Quaint. This breaks MSSQL.